src/main/java/backend/resource/TurboMilestone.java

```diff
@@ -83,6 +86,60 @@ public TurboMilestone(TurboMilestone milestone) {
     private void ______METHODS______() {
     }
 
+    public boolean isOverdue() {
+        return dueDate.isPresent() && dueDate.get().isBefore(LocalDate.now());
+    }
+
+    public boolean hasOpenIssues() {
+        return openIssues > 0;
+    }
+
+    /**
+     * A milestone is ongoing if it is open and not due yet or if it is overdue but still open and has open issues.
+     */
+    public boolean isOngoing() {
+        return isOpen() && (!isOverdue() || hasOpenIssues());
+    }
+
+    public static List<TurboMilestone> filterMilestonesOfRepos(List<TurboMilestone> milestones,
+                                                               List<String> repoIds) {
+        return milestones.stream()
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+    }
+
+    public static List<TurboMilestone> getOpenMilestones(List<TurboMilestone> milestones) {
+        return milestones.stream()
+                .filter(TurboMilestone::isOpen)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Condition: milestone must have due dates
+     */
+    public static Comparator<TurboMilestone> getDueDateComparator() {
+        return (a, b) -> {
+            assert a.getDueDate().isPresent();
+            assert b.getDueDate().isPresent();
+            LocalDate aDueDate = a.getDueDate().get();
+            LocalDate bDueDate = b.getDueDate().get();
+            return aDueDate.compareTo(bDueDate);
+        };
+    }
+
+    public static List<TurboMilestone> sortByDueDate(List<TurboMilestone> milestones) {
```

Comment:
As this is a public method, you may want to mention what happens if multiple milestones have the same due date.